webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PolicyWriterService.scala

```diff
@@ -848,9 +848,10 @@ class PolicyWriterServiceImpl(
       val writeAgent = for {
         // changing `writeAllAgentSpecificFiles.write` to IOResult breaks DSC
         t0 <- currentTimeNanos
-        _  <- IOResult
-                .attempt(writeAllAgentSpecificFiles.write(prepared))
+        _  <- writeAllAgentSpecificFiles
+                .write(prepared)
                 .chainError(s"Error with node '${prepared.paths.nodeId.value}'")
+                .toIO
```

Comment:
The `write` method should not throw at all, all implementations have already been rewritten with `PureResult` instead